### ISSUE TITLE
the `and` macro does not produce the return statement

### ISSUE THREAD

**Issue by akond** [squint-cljs/squint#352]
This code:
```(defn inject-links! []
  (-> (js/$ "main > div > div")
    (.eq 1)
    (.find "*")
    (.filter (fn [_ element]
           (and
             (not= "A" (.-nodeName element)) ; <------------THIS PART HERE
             (-> (js/$ element) (.parents "a") .-length zero?))))
    (.contents)
    (.filter (fn [_ element]
           (= 3 (.-nodeType element))))
    (.each (fn [_ text]
           (.appendData text "+")))))
```

gets compiled down to

```var inject_links_BANG_ = (function () {
return $("main > div > div").eq(1).find("*").filter((function (_, element) {
let and__25548__auto__1 = ("A" !== element["nodeName"]);
if (and__25548__auto__1 != null && and__25548__auto__1 !== false) {
($(element).parents("a")["length"] == 0)} else { // <------------  PRODUCES NO RETURN STATEMENT HERE
return and__25548__auto__1;}
})).contents().filter((function (_, element) {
return (3 === element["nodeType"]);
})).each((function (_, text) {
return text.appendData("+");
}));
});
```

The result is this function returns `undefined` when the first part of the `and` expression is true.

**Comment by borkdude** [comment]
I am trying to find a repro that I can run without any jquery stuff, so I can repro locally:

This seems to work:

``` clojure
(-> [1 2 3]
    (.filter (fn [_ element]
               (and
                (not= 1 element)
                (not (zero? element)))))
    prn)
```

Is it possible to adapt this example to repro your problem above?

**Comment by borkdude** [comment]
Also please specify your squint version

**Comment by borkdude** [comment]
Found a repro:

```
(->
    #js [0 1 2]
    (.filter (fn [element _]
               (and
                (not= 1 element)
                (zero? element))))
    prn)
```

This should return `#js [0]` (as it does with normal CLJS) but returns an empty array in squint.

**Comment by borkdude** [comment]
Fixed in 0.3.36!

**Comment by akond** [comment]
It works! Thank you very much.
